### PR TITLE
Bugfix/gitlab change permission for all programming exericses

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added `@PreAuthorize` and checked the course groups for all new REST Calls (security).
- [x] I implemented the changes with a good performance and prevented too many database calls.
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

Quick fix for #4081

### Description

Tutors are now added to exam exercises in GitLab as well

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Tutor

1. Create an exam with a programming exercise in an environment that uses GitLab
2. Add a new user as tutor in the course
3. Check that the new tutor can access the repositories of the exercise in GitLab

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage
No changes

### CODE REVIEW

**Review by JohannesStoehr — COMMENTED** [review]
Tested with Florian and Niclas on TS2 and reviewed the code, which looks good.
One thing that might need some refactoring in a followup: Currently the role name and the role enum are used as parameters. This might lead to unintentional bugs. I'd suggest only passing the role enum and retrieving the corresponding group name in the method.

**Review by niclasschuemann — APPROVED** [review]

File: src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java [hunk 1]
```diff
@@ -1365,7 +1368,7 @@ public void fetchParticipationsWithSubmissionsAndResultsForCourses(List<Course>
     public ResponseEntity<Void> removeInstructorFromCourse(@PathVariable Long courseId, @PathVariable String instructorLogin) {
         log.debug("REST request to remove {} as instructor from course : {}", instructorLogin, courseId);
         var course = courseRepository.findByIdElseThrow(courseId);
-        return removeUserFromCourseGroup(instructorLogin, userRepository.getUserWithGroupsAndAuthorities(), course, course.getInstructorGroupName());
+        return removeUserFromCourseGroup(instructorLogin, userRepository.getUserWithGroupsAndAuthorities(), course, course.getInstructorGroupName(), Role.STUDENT);
```
Comment (JohannesStoehr) [inline src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java:1371]: This should probably be Role.Instructor

File: src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java [hunk 2]
```diff
@@ -1320,7 +1323,7 @@ public void fetchParticipationsWithSubmissionsAndResultsForCourses(List<Course>
     public ResponseEntity<Void> removeStudentFromCourse(@PathVariable Long courseId, @PathVariable String studentLogin) {
         log.debug("REST request to remove {} as student from course : {}", studentLogin, courseId);
         var course = courseRepository.findByIdElseThrow(courseId);
-        return removeUserFromCourseGroup(studentLogin, userRepository.getUserWithGroupsAndAuthorities(), course, course.getStudentGroupName());
+        return removeUserFromCourseGroup(studentLogin, userRepository.getUserWithGroupsAndAuthorities(), course, course.getStudentGroupName(), Role.INSTRUCTOR);
```
Comment (JohannesStoehr) [inline src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java:1326]: That should probably be Role.Student

**Review by niclasschuemann — CHANGES_REQUESTED** [review]
Johannes comments should be addressed in some way, we should at least keep them in mind for a follow-up pr

**Review by FlorianGlombik — APPROVED** [review]
Tested it on ts2, an existing user was added to a course as a tutor after the end of an exam. The tutor got the respective access rights (reporter) for an existing exam exercise.
Once the two comments https://github.com/ls1intum/Artemis/pull/4082#discussion_r721673983 https://github.com/ls1intum/Artemis/pull/4082#discussion_r721674416 are addressed this bugfix can be merged.

**Review by niclasschuemann — DISMISSED** [review]

**Comment by krusche** [comment]
> One thing that might need some refactoring in a followup: Currently the role name and the role enum are used as parameters. This might lead to unintentional bugs. I'd suggest only passing the role enum and retrieving the corresponding group name in the method.

I agree, let's fix this in a follow-up, when the enums are actually used. We can then introduce a method in Course.java to retrieve the group name based on the role enum

**Pull request merged**